test.js

```diff
@@ -291,14 +350,173 @@ test('Does not increment maxListeners of merged streams', async t => {
 	const inputStreams = Array.from({length}, () => Readable.from(['.']));
 	const stream = mergeStreams(inputStreams);
 	assertMaxListeners(t, stream, defaultMaxListeners);
+
+	await stream.toArray();
+	await scheduler.yield();
+	t.is(stream.getMaxListeners(), defaultMaxListeners);
+});
+
+test('Updates maxListeners of merged streams with add() and remove()', async t => {
+	const stream = mergeStreams([Readable.from('.')]);
+	assertMaxListeners(t, stream, defaultMaxListeners);
+
+	const inputStream = Readable.from('.');
+	stream.add(inputStream);
+	assertMaxListeners(t, stream, defaultMaxListeners);
+
+	stream.remove(inputStream);
+	await scheduler.yield();
+	assertMaxListeners(t, stream, defaultMaxListeners);
+
 	await stream.toArray();
+	await scheduler.yield();
+	t.is(stream.getMaxListeners(), defaultMaxListeners);
+});
+
+test('Handles setting maxListeners to Infinity', async t => {
```

Comment:
Some of the test titles start with capital letter and some don't. Would be nice with some consistency.

Reply:
Fixed in 214f7907e79fac37966d2277d2951ae24e425946.